### PR TITLE
Simple (even too simple) door locking

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Features "You can close the doors now. How did you live your life without this"
#### Purpose of change
Reddit annoyed me enough so i tried to make something useful using #64661
#### Describe the solution
Doors has an `examine_action`, that cast EoC, EoC checks if you inside, transform any door around you into locked, print the message and assign 2 second activity
#### Problems
`examine_action` do not support range - can't make door closing action for doors that already have `door_peephole` action
Locked every door in the radius, not exactly one that you examine
Can't lock the door by using `c`lose button, need to `e`xamine it
~~there is no locked metal door with peephole~~
#### Testing
GIF 14-04-2023 0-22-04
#### Additional context
are you happy, Reddit?

### CODE REVIEW

**Comment by TheShadowFerret** [comment]
~~Oh boy lockpick practice~~

File: data/json/eocs/door_locking.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      { "u_location_variable": { "global_val": "door_to_close" }, "min_radius": 1, "max_radius": 1 },
+      {
+        "u_transform_radius": 1,
+        "ter_furn_transform": "lock_the_door",
+        "target_var": { "global_val": "door_to_close" }
+      },
+      { "u_assign_activity": "ACT_GENERIC_EOC", "duration": "2 seconds" },
+      { "u_message": "You rotate the deadbolt lock, and close the door from the inside." }
+    ],
+    "false_effect": { "u_message": "Can't close the door from the inside." }
```
Comment (Maleclypse) [inline data/json/eocs/door_locking.json:16]:
```suggestion
    "false_effect": { "u_message": "Can't lock this door from the outside." }
```

**Review by Maleclypse — COMMENTED** [review]
I think the false effect only fires if you are outside?

**Comment by GuardianDll** [comment]
Right

File: data/json/eocs/door_locking.json (hunk 1 above)
Comment (GuardianDll) [reply data/json/eocs/door_locking.json]: ~~that's why you don't make a prs at 1 am~~

**Comment by John-Candlebury** [comment]
Wouldnt a system that requires no micromanagement be better?
You interact with a lockable door once and whenever it gets closed after that it cant be opened by others forevermore (basically default behavior changes to close + lock door whenever you close it)

**Comment by GuardianDll** [comment]
I am not sure what i can do personally that would work better than this

**Comment by Broken27** [comment]
Yeah i am pretty sure this won't get merged.

This doesn't solve the issues Kevin had with #61087

**Comment by GuardianDll** [comment]
Let's speed it up

**Pull request closed without merging**